cmd/axelard/cmd/vald/start.go

```diff
@@ -320,30 +323,47 @@ func createJob(sub tmEvents.FilteredSubscriber, processor func(event tmEvents.Ev
 
 }
 
-func getStartBlock(stateStore StateStore, tmClient tmEvents.BlockHeightClient, logger log.Logger) (int64, error) {
+// Get the block height to start listening for TM events from.
+// Checks that the node is not too out of sync from the network.
+// If state.json is not stale, uses that as starting height.
+// Otherwise, starts from the node height.
+func getStartBlock(cfg config.ValdConfig, stateStore StateStore, tmClient tmEvents.BlockHeightClient, logger log.Logger) (int64, error) {
 	startBlock, err := stateStore.GetState()
 	if err != nil {
-		logger.Error(err.Error())
-
-		return 0, nil
+		logger.Info(err.Error())
+		startBlock = 0
+	} else {
+		startBlock++ // Skip the block that might have already been executed
 	}
 
 	rpcCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	height, err := tmClient.LatestBlockHeight(rpcCtx)
+	nodeHeight, err := tmClient.LatestNodeBlockHeight(rpcCtx)
+	if err != nil {
+		return 0, err
+	}
+
+	networkHeight, err := tmClient.LatestBlockHeight(rpcCtx)
```

Comment:
I can't seem to find any doc on what the rpc method `abci_info` really does?

Reply:
https://docs.tendermint.com/master/spec/abci/abci.html#info

Reply:
I found that and it really doesn't explain how does it figure out what the network height is.

Reply:
Yeah, I'm looking into it, and it sounds like it doesn't do anything different :/ Pretty sure I used this to get the latest network height, but maybe I was doing something wrong. Let's wait on this then. I'll verify again. If this doesn't get the latest, then I can switch to checking the `catching_up` flag in the query response instead.

Reply:
@milapsheth sync a brand new node on testnet and query `status` and `abci_info` and compare the results?